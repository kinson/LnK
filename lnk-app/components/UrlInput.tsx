import React, { ReactElement } from 'react';

interface Props {
  hidden: boolean;
  url: string | null;
  updateUrl: (res: React.FormEvent<HTMLInputElement>) => void;
}

export default function UrlInput({
  hidden,
  url,
  updateUrl,
}: Props): ReactElement {
  const baseStyle =
    'placeholder-gray-300 text-darkPurple dark:text-white text-md relative bg-white dark:bg-gray-600 rounded border-0 shadow outline-none focus:outline-none w-full transition-opacity duration-300';

  const style = hidden ? `${baseStyle} h-0 p-0 opacity-0` : `${baseStyle} p-3`;

  return (
    <input
      value={url ?? ''}
      onChange={updateUrl}
      name="url-input"
      id="url-input"
      type="text"
      placeholder="https://google.com/?q=A+Long+Query+You+Want+To+Share+With..."
      className={style}
    />
  );
}
