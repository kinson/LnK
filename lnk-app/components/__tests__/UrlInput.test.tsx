import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';
import UrlInput from '../UrlInput';

describe('UrlInput component tests', () => {
  it('should render the UrlInput component when it not hidden', () => {
    const component = renderer.create(
      <UrlInput hidden={false} url="https://..." updateUrl={jest.fn()} />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should hide the UrlInput component when it is hidden', () => {
    const component = renderer.create(
      <UrlInput hidden url="https://..." updateUrl={jest.fn()} />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should call the updateUrl prop when changed', async () => {
    const updateUrlMock = jest.fn();
    const component = render(
      <UrlInput hidden={false} url="https://..." updateUrl={updateUrlMock} />
    );

    const inputElement = await component.findByPlaceholderText(
      'https://google.com/?q=A+Long+Query+You+Want+To+Share+With...'
    );

    fireEvent.change(inputElement, { target: { value: 'a' } });

    expect(updateUrlMock).toHaveBeenCalled();
  });
});
