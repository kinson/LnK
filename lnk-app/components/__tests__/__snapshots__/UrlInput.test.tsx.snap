// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrlInput component tests should hide the UrlInput component when it is hidden 1`] = `
<input
  className="placeholder-gray-300 dark:placeholder-gray-500 text-darkPurple dark:text-white text-md relative bg-white dark:bg-gray-600 rounded border-0 shadow outline-none focus:outline-none w-full transition-opacity duration-300 h-0 p-0 opacity-0"
  id="url-input"
  name="url-input"
  onChange={[MockFunction]}
  onKeyPress={[Function]}
  placeholder="https://google.com/?q=A+Long+Query+You+Want+To+Share+With..."
  type="text"
  value="https://..."
/>
`;

exports[`UrlInput component tests should render the UrlInput component when it not hidden 1`] = `
<input
  className="placeholder-gray-300 dark:placeholder-gray-500 text-darkPurple dark:text-white text-md relative bg-white dark:bg-gray-600 rounded border-0 shadow outline-none focus:outline-none w-full transition-opacity duration-300 p-3"
  id="url-input"
  name="url-input"
  onChange={[MockFunction]}
  onKeyPress={[Function]}
  placeholder="https://google.com/?q=A+Long+Query+You+Want+To+Share+With..."
  type="text"
  value="https://..."
/>
`;
